lib/elastomer/client/docs.rb

```diff
@@ -273,47 +273,10 @@ def count( query, params = nil )
       # See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-delete-by-query.html
       #
       # Returns a Hash of statistics about the delete operations
-      def delete_by_query( query, params = nil )
+      def delete_by_query(query, params = nil)
         query, params = extract_params(query) if params.nil?
 
-        responses = []
-        accumulate = lambda { |response| responses << response unless response == nil }
-
-        # accumulate is called both inside and outside the bulk block in order
-        # to capture bulk responses returned from calls to `delete` and the call
-        # to `bulk`
-        accumulate.call(bulk(params) do |b|
-          scan(query, params).each_document do |hit|
-            accumulate.call(b.delete(_id: hit["_id"], _type: hit["_type"], _index: hit["_index"]))
-          end
-        end)
-
-        # collects the array of responses containing arrays of delete action
-        # hashes into an array
-        response_items = responses.flat_map { |r| r["items"].map { |i| i["delete"] } }
-
-        is_ok = lambda { |status| status.between?(200, 299) }
-
-        categorize = lambda do |items|
-          {
-            "found" => items.count { |i| i["found"] },
-            "deleted" => items.count { |i| is_ok.call(i["status"]) },
-            "missing" => items.count { |i| !i["found"] },
-            "failed" => items.count { |i| i["found"] && !is_ok.call(i["status"]) },
-          }
-        end
-
-        indices = Hash[response_items
-          .group_by { |i| i["_index"] } # indexes the delete hashes by index name
-          .map { |index, items| [index, categorize.call(items)] }]
-
-        indices_with_all = indices.merge({ "_all" => indices.values.reduce({}) { |acc, i| acc.merge(i) { |_, n, m| n + m } } })
-
-        {
-          "took" => responses.map { |r| r["took"] }.reduce(:+),
-          "_indices" => indices_with_all,
-          "failures" => response_items.select { |i| !is_ok.call(i["status"]) },
-        }
+        @client.delete_by_query(query, update_params(params, { :index => @name, :type => @type }))
```

Comment:
No need to override `:index` and `:type` here, they'll be added by `Docs#defaults`.